lib/handler/connect.c

```diff
@@ -910,26 +952,41 @@ static void on_generator_dispose(void *_self)
     dispose_generator(self);
 }
 
-static int on_req(h2o_handler_t *_handler, h2o_req_t *req)
+/**
+ * expects "/host/port/" as input, where the preceding slash is optional
+ */
```

Comment:
Exactly!

The second argument passed to `masque_decode_hostport` is `req->path_normalized.base + req->pathconf->path.len`.

The registration (i.e., `h2o_req_t::path_normalized`) may or may not include `/`. This is IIUC tradition of  HTTP servers to allow either one exact path be mapped to a specific handler, or to explicitly map a directory to a directory handler.

In case of the CONNECT handler, we want to run correctly regardless of trailing slash being specified, hence this design.